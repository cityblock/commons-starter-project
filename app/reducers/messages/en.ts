import { Lang } from '../locale-reducer';

export const ENGLISH_TRANSLATION = {
  lang: 'en' as Lang,
  messages: {
    'lightbox.of': 'of',
    'lightbox.image': 'image',
    'lightbox.images': 'images',
    'login.logInGoogle': 'Sign in with Google',
    'patientPanel.header': 'Patient Panel',
    'patientPanel.addPatient': 'Add Patient',
    'patientPanel.firstName': 'First Name',
    'patientPanel.lastName': 'Last Name',
    'patientPanel.age': 'Age',
    'patientPanel.location': 'Location',
    'patientPanel.joinedAt': 'Patient Joined',
    'patientPanel.engagedAt': 'Last Update',
    'patient.dateOfBirth': 'Date of birth',
    'patient.language': 'Preferred language',
    'patient.location': 'Location',
    'patient.joinedAt': 'Patient since',
    'patient.tasks': 'Tasks',
    'patient.timeline': 'Timeline',
    'patient.threeSixty': '360° View',
    'patient.map': 'MAP',
    'patient.context': 'Context',
    'patient.activity': 'Activity',
    'patient.submitProgressNote': 'Sign and Submit',
    'patient.activeCarePlan': 'Active',
    'patient.inactiveCarePlan': 'Inactive',
    'patient.medicaid': 'Medicaid',
    'patient.carePlanSuggestions': 'Suggestions',
    'patient.cbhNumber': 'CBH MRN#',
    'patient.patientInfo': 'Patient Info',
    'patient.createdPopupHeading': 'successfully enrolled',
    'patient.createdPopupBody':
      'Continue to their patient profile or go back to your patient roster.',
    'patient.createdPopupRoster': 'Patient Roster',
    'patient.createdPopupProfile': 'Go to profile',
    'encounter.new': 'Record new encounter',
    'encounter.newReason': 'Encounter reason',
    'encounter.newLocation': 'Location of encounter',
    'encounter.newStartTime': 'Start time',
    'encounter.newSummary': 'Encounter summary',
    'encounter.newAttachment': 'Add attachment',
    'encounter.newCancel': 'Cancel',
    'encounter.newSubmit': 'Submit',
    'error.tableLoadingHeading': 'Unable to load your roster',
    'error.tableLoadingMessage':
      'Sorry, something went wrong. Please try reloading the page again.',
    'error.tableLoadingButton': 'Try Again',
    'error.tableLoading': 'Loading',
    'backLink.back': 'Back',
    'careTeam.slack': 'Message on Slack',
    'careTeam.email': 'Send an email',
    'computedField.flag': 'Flag for review',
    'computedField.flagDetail':
      'Provide the following information for this question to be reviewed',
    'computedField.reason': 'Reason for review:',
    'computedField.reasonDetail': 'Enter the reason for review...',
    'concernCreate.addConcern': 'Add concern',
    'concernCreate.cancel': 'Cancel',
    'concernCreate.detail': "Complete this form to add a concern to this member's MAP",
    'concernCreate.hideAll': 'Hide all concerns',
    'concernCreate.loading': 'Loading...',
    'concernCreate.noResults': 'No matching concerns found.',
    'concernCreate.placeholder': 'Search for a concern...',
    'concernCreate.selectLabel': 'Add a concern:',
    'concernCreate.showAll': 'See all possible concerns',
    'concernCreate.submit': 'Add concern',
    'concernCreate.title': 'Add a concern',
    'concernCreate.selectConcernType': 'Select a concern type',
    'concernCreate.selectConcernTypeLabel': 'Concern type',
    'concernDelete.description':
      'Deleting it will completely remove it from this patient’s record.',
    'concernDelete.menu': 'Delete Concern',
    'concernDelete.name': 'Concern to be deleted:',
    'concernDelete.title': 'Are you sure you want to delete this concern?',
    'concernStats.created': 'Created:',
    'concernStats.goals': 'Goals:',
    'concernStats.lastUpdated': 'Last update:',
    'concernStats.tasks': 'Tasks:',
    'dashboard.countLoading': 'Loading',
    'dashboard.lists': 'Member lists',
    'dashboard.notifications': 'Notifications:',
    'dashboard.noUrgentTasks': 'There are no patients with tasks due or notifications',
    'dashboard.suggestions': 'Pending MAP suggestions',
    'dashboard.tasks': 'Tasks due and notifications',
    'dashboard.tasksDue': 'Tasks due:',
    'dashboard.team': 'New to your care team',
    'forms.optional': 'optional',
    'forms.declined': 'Declined',
    'forms.yes': 'Yes',
    'forms.no': 'No',
    'forms.clickHere': 'click here',
    'dateInfo.created': 'Created:',
    'dateInfo.updated': 'Updated:',
    'demographicsForm.dateOfBirth': 'Date of birth',
    'demographicsForm.firstName': 'First name',
    'demographicsForm.middleName': 'Middle name',
    'demographicsForm.lastName': 'Last name',
    'demographicsForm.maritalStatus': 'Marital status',
    'demographicsForm.maritalStatusPlaceholder': 'Select status',
    'demographicsForm.language': 'Preferred language',
    'demographicsForm.languagePlaceholder': 'Select language',
    'demographicsForm.gender': 'Gender',
    'demographicsForm.genderPlaceholder': 'Select gender',
    'demographicsForm.race': 'Race',
    'demographicsForm.racePlaceholder': 'Select race',
    'demographicsForm.ssn': 'Social security number',
    'demographicsForm.zip': 'Zip code',
    'contactForm.email': 'Email address',
    'contactForm.homePhone': 'Home phone number',
    'contactForm.mobilePhone': 'Mobile phone number',
    'contactForm.consentToText': 'Does the patient consent to being contacted via text?',
    'contactForm.consentToPhone': 'Does the patient consent to being contacted via phone?',
    'contactForm.readConsentStatement': 'Read consent statement to patient:',
    'screeningTool.submit': 'Submit',
    'screeningTool.resultsTitle': 'New Care Plan Suggestions',
    'screeningTool.resultsBody':
      "Based on the results of this tool, the following have been recommended as additions to the patient's care plan.",
    'goalCreate.addGoal': 'Add a goal',
    'goalCreate.back': 'Back',
    'goalCreate.cancel': 'Cancel',
    'goalCreate.custom': 'Write a new goal',
    'goalCreate.detail': "Choose from a list of goals or write a custom goal for this member's MAP",
    'goalCreate.goalAdded': 'Goal added:',
    'goalCreate.hideAll': 'Hide list of goals',
    'goalCreate.loading': 'Loading...',
    'goalCreate.noResults': 'No matching goals found. Adding as a custom goal.',
    'goalCreate.search': 'Search for a goal or write a custom goal...',
    'goalCreate.selectGoal': 'Select goal',
    'goalCreate.selectLabel': 'Add a goal:',
    'goalCreate.showAll': 'See list of goals',
    'goalCreate.submit': 'Add goal',
    'goalCreate.submitWithTasks': 'Done',
    'goalCreate.suggestedTasks': 'Suggested tasks:',
    'goalCreate.suggestionsDetail':
      'Based on this goal addition, the following suggestions have been recommended to this Member Action Plan. You must accept or dismiss these suggestions before continuing.',
    'goalCreate.suggestionsTitle': 'New MAP suggestions',
    'goalCreate.templates': 'All other goals',
    'goalCreate.title': 'Enter goal title:',
    'goalCreate.titlePlaceholder': 'Enter goal title...',
    'goalDelete.description': 'Deleting it will completely remove it from this patient’s record.',
    'goalDelete.menu': 'Delete Goal',
    'goalDelete.name': 'Goal to be deleted:',
    'goalDelete.title': 'Are you sure you want to delete this goal?',
    'history360.administered': 'Administered by:',
    'history360.conducted': 'Conducted:',
    'history360.noRecord': 'No record',
    'history360.previous': 'Previous score:',
    'insuranceForm.insuranceType': 'Insurance type',
    'insuranceForm.insuranceTypePlaceholder': 'Select insurance type',
    'insuranceForm.policyHolderRelationship': 'Patient relationship to policyholder',
    'insuranceForm.policyHolderRelationshipPlaceholder': 'Select relationship',
    'insuranceForm.memberId': 'Member ID',
    'insuranceForm.policyGroupNumber': 'Policy group number',
    'insuranceForm.issueDate': 'Issue date',
    'insuranceForm.expirationDate': 'Expiration date',
    'modalButtons.cancel': 'Cancel',
    'modalButtons.delete': 'Yes, delete',
    'modalButtons.submit': 'Submit',
    'patientInfo.jumpTo': 'Jump to:',
    'patientInfo.saveChanges': 'Save changes',
    'patientInfo.demographicInfo': 'Demographic information',
    'patientInfo.contactInfo': 'Contact information',
    'patientInfo.insuranceInfo': 'Insurance information',
    'patientMap.addGoal': 'Add a new goal',
    'patientMap.addTask': 'Add a task',
    'patientMap.nextUp': 'Next Up',
    'patientMap.emptyNextUpHeader': 'There are no Next Up concerns for this patient',
    'patientMap.emptyNextUpDetail':
      'Add concerns here that the patient and the care team do not want to focus on right now but would like to keep track of',
    'patientSearch.address': 'Home address',
    'patientSearch.dateOfBirth': 'Date of birth',
    'patientSearch.memberId': 'Member ID',
    'patientSearch.name': 'Patient name',
    'patientSearch.noResults': 'No results for this search',
    'patientSearch.noResultsDetail':
      'No matching patients for this search. Please check your spelling or try a new search.',
    'patientSearch.of': 'of',
    'patientSearch.placeholder': 'Search for member...',
    'patientSearch.resultsDescription': 'members found',
    'patientSearch.resultsDescriptionSingle': 'member found',
    'patientSearch.resultsTitle': 'Search results:',
    'patientSearch.resultsPlaceholder': 'Search for patients on your care team',
    'patientSearch.search': 'Search',
    'patientSearch.searchDescription': 'Search by first name or last name',
    'patientSearch.searchTitle': 'Search for member',
    'patientSearch.status': 'Status',
    'header.settings': 'Settings',
    'header.search': 'Search',
    'header.patients': 'Members',
    'header.tasks': 'Tasks',
    'header.logOut': 'Logout',
    'header.builder': 'Builder',
    'header.manager': 'Manager',
    'riskArea.assessmentType': 'Select assessment type',
    'riskArea.automated': 'Automated Assessment',
    'riskArea.highRiskThreshold': 'Enter assessment high risk threshold',
    'riskArea.manual': 'Manual Assessment',
    'riskArea.mediumRiskThreshold': 'Enter assessment medium risk threshold',
    'riskArea.order': 'Enter assessment order',
    'riskArea.title': 'Enter assessment title',
    'riskAreaGroup.close': 'Close',
    'riskAreaGroup.create': 'Create Domain',
    'riskAreaGroup.delete': 'Delete Domain',
    'riskAreaGroup.deleteDetail': 'Domain to be deleted:',
    'riskAreaGroup.deleteWarning': 'Are you sure you want to delete this domain?',
    'riskAreaGroup.empty': 'No Domains',
    'riskAreaGroup.highRiskThreshold': 'Enter domain high risk threshold',
    'riskAreaGroup.mediumRiskThreshold': 'Enter domain medium risk threshold',
    'riskArea.riskAreaGroupId': 'Select Domain',
    'riskAreaGroup.order': 'Enter domain order',
    'riskAreaGroup.shortTitle': 'Enter short domain title (radar chart)',
    'riskAreaGroup.title': 'Enter domain title',
    'riskAreaAssessment.administer': 'Administer tools',
    'riskAreaAssessment.save': 'Done',
    'riskAreaAssessment.start': 'Edit answers',
    'riskAreaAssessment.resultsTitle': 'New Care Plan Suggestions',
    'riskAreaAssessment.resultsBody':
      "Based on the results of this assessment, the following have been recommended as additions to the patient's care plan.",
    'select.default': 'Select one',
    'settings.heading': 'Settings',
    'settings.save': 'Save',
    'tasks.assignedToPlaceholder': 'Assign to',
    'tasks.patientGoalPlaceholder': 'Patient goal',
    'tasks.listView': 'Tasks',
    'tasks.createTask': 'Create a new task',
    'tasks.notifications': 'Notifications',
    'tasks.noTasks': 'No Tasks',
    'taskCreate.addTask': 'Add a task',
    'taskCreate.assignee': 'Assign task to:',
    'taskCreate.cancel': 'Cancel',
    'taskCreate.concern': 'Concern:',
    'taskCreate.description': 'Enter task description:',
    'taskCreate.descriptionPlaceholder': 'Enter task description...',
    'taskCreate.detail': 'Complete this form to add a task to this goal',
    'taskCreate.dueAt': 'Due date:',
    'taskCreate.goal': 'Goal:',
    'taskCreate.priority': 'Task priority:',
    'taskCreate.submit': 'Add task',
    'taskCreate.title': 'Enter task title:',
    'taskCreate.titlePlaceholder': 'Enter task title...',
    'taskComment.activity': 'Activity and comments',
    'taskComment.error': 'Error adding comment',
    'taskDelete.cancel': 'Cancel',
    'taskDelete.confirm': 'Yes, delete',
    'taskDelete.title': 'Are you sure you want to delete this task?',
    'taskDelete.titleError': 'Error deleting task.',
    'taskDelete.body': "Deleting this task will completely remove it from this patient's record.",
    'taskDelete.bodyError': 'Please try again.',
    'taskDescription.empty': 'Click to add a task description...',
    'task.assign': 'Assign Task:',
    'task.due': 'Due',
    'task.followers': 'Followers of this task:',
    'task.opened': 'Opened',
    'task.noDueDate': 'No due date',
    'task.priority': 'Task priority:',
    'taskPriority.low': 'Low priority',
    'taskPriority.medium': 'Medium priority',
    'taskPriority.high': 'High priority',
    'taskPriority.select': 'Select priority',
    'textarea.default': 'Enter response...',
    'threeSixty.automated': 'Automated Assessments:',
    'threeSixty.automatedDetail': 'Automated Assessments',
    'threeSixty.automatedEmpty': 'There are no automated assessments on record',
    'threeSixty.back': 'Back to 360',
    'threeSixty.history': 'History',
    'threeSixty.historyEmpty': 'No screening tools have been administered to this patient',
    'threeSixty.initialAssessment': 'Initial Assessment:',
    'threeSixty.manual': 'Manual Assessments:',
    'threeSixty.manualDetail': 'Manual Assessments',
    'threeSixty.manualEmpty': 'There are no manual assessments on record',
    'threeSixty.never': 'Not submitted',
    'threeSixty.noAssessments': 'No assessments on record',
    'threeSixty.notCompleted': 'Member has not completed this assessment.',
    'threeSixty.notCompletedShort': 'Not completed',
    'threeSixty.summary': 'Patient Summary',
    'notifications.leftPane': 'Left Pane',
    'notifications.noNotifications': 'No Notifications',
    'notification.time': 'Time',
    'user.createdAt': 'Created At',
    'user.editedAt': 'Last edited',
    'user.delete': 'Delete User',
    'progressNote.new': 'New progress note',
    'progressNote.progressNotes': 'Progress Note',
    'progressNote.selectType': 'Select progress note type:',
    'progressNote.selectTime': 'Select progress note time:',
    'progressNote.selectLocation': 'Select progress note location:',
    'progressNote.contextAndPlan': 'Context and plan',
    'progressNote.memberConcernAndObservation': 'Member concern and observation',
    'progressNote.numberMapUpdates': 'MAP updates',
    'progressNote.goToActivity': 'go to activity',
    'progressNote.updateMap': 'Update MAP',
    'progressNote.update360': 'Update 360',
    'progressNote.administerTool': 'Administer tool',
    'progressNote.doesRequireCosignature': 'Does this progress note require supervisor review?',
    'progressNote.selectSupervisor': 'Select supervisor:',
    'progressNote.supervisorReview': 'Supervisor Note',
    'progressNote.pendingReview': 'Pending review',
    'progressNote.reviewed': 'Reviewed',
    'quickCallNote.new': '+ Quick call note',
    'quickCallForm.title': 'Quick call note',
    'quickCallForm.submit': 'Save call',
    'quickCallForm.inboundCall': 'Inbound call',
    'quickCallForm.outboundCall': 'Outbound call',
    'quickCallForm.startTime': 'What time was the call?',
    'quickCallForm.callRecipient': 'Who was the call with?',
    'quickCallForm.wasSuccessful': 'Were you able to speak with them?',
    'quickCallForm.reason': 'What was the reason for the call?',
    'quickCallForm.summary': 'Please provide additional notes about the call here …',
    'computedField.create': 'Create Computed Field',
    'computedFieldCreate.createLabelPlaceholder': 'Enter label',
    'computedFieldCreate.createDataTypeLabel': 'Select a data type',
    'computedFieldCreate.cancel': 'Cancel',
    'computedField.confirmDelete': 'Yes, delete',
    'question.selectComputedField': 'Set computed field',
    'question.notComputedField': 'Not a computed field',
    'question.selectApplicable': 'Select an applicable if type (required!)',
    'question.applicableOneTrue': 'One true',
    'question.applicableAllTrue': 'All true',
    'question.selectAnswerType': 'Select an answer type (required!)',
    'question.answerTypeDropdown': 'Dropdown',
    'question.answerTypeRadio': 'Radio',
    'question.answerTypeFreeText': 'Free Text',
    'question.answerTypeMultiselect': 'Multiselect',
    'question.selectHasOtherTextAnswer': 'Should this question have an "other" answer?',
    'question.hasOtherTextAnswerTrue': 'Yes',
    'question.hasOtherTextAnswerFalse': 'No',
    'questionConditionCreate.selectAnswer': 'Select Answer',
    'carePlanSuggestions.seeSuggestions': 'See Suggestions',
    'riskAdjustmentType.inactive': 'inactive',
    'riskAdjustmentType.increment': 'increment',
    'riskAdjustmentType.forceHighRisk': 'force high risk',
    'riskAdjustmentType.selectRiskAdjustmentType': 'Select risk adjustment type',
    'builder.createConcern': 'Create Concern',
    'builder.createAssessment': 'Create Assessment',
    'builder.createQuestion': 'Create Question',
    'builder.createGoal': 'Create Goal',
    'builder.createProgressNoteTemplate': 'Create Progress Note Template',
    'builder.domains': 'Domains',
    'builder.assessments': 'Assessments',
    'builder.questions': 'Questions',
    'builder.concerns': 'Concerns',
    'builder.goals': 'Goals',
    'builder.tools': 'Tools',
    'builder.progressNoteTemplates': 'Progress Note Templates',
    'builder.computedFields': 'Computed Fields',
    'builder.createScreeningTool': 'Create Screening Tool',
    'manager.inviteUser': 'Invite User',
    'manager.users': 'Users',
    'manager.invites': 'Invites',
  },
};
