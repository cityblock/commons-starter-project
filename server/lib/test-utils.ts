import Knex from 'knex';
import omit from 'lodash/omit';
import Objection from 'objection';
import * as knexConfig from '../models/knexfile';

export const setupDb = () => {
  const config = (knexConfig as { [key: string]: any }).test;
  const knex = Knex(config);
  Objection.Model.knex(knex);
  return {
    destroy: () => {
      knex.destroy();
    },
  };
};

export const testGraphqlContext = (ctx: any) => ({
  getOrCreateTransaction: (
    testTransaction: Objection.Transaction,
    cb: <T>(cb: Objection.Transaction) => Promise<T>,
  ) => cb(testTransaction),
  ...ctx,
});

export const filterTimestamps = (object: { createdAt?: any, updatedAt?: any, deletedAt: any }) => 
  omit(object, 'createdAt', 'updatedAt', 'deletedAt');
