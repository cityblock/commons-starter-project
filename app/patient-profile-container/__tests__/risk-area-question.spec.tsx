import { createMemoryHistory } from 'history';
import * as React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { ConnectedRouter } from 'react-router-redux';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { ENGLISH_TRANSLATION } from '../../reducers/messages/en';
import ReduxConnectedIntlProvider from '../../redux-connected-intl-provider';
import RiskAreaQuestion from '../risk-area-question';

const locale = { messages: ENGLISH_TRANSLATION.messages };
const mockStore = configureMockStore([]);

it('renders a risk area question', () => {
  const history = createMemoryHistory();
  const question = {
    id: '123',
    createdAt: 'Wed Aug 02 2017 18:31:45 GMT-0400 (EDT)',
    deletedAt: null,
    title: 'Question',
    answerType: 'multiselect' as any,
    riskAreaId: '321',
    order: 1,
    validatedSource: 'Duke Population Health',
    answers: [{
      id: '456',
      displayValue: 'Answer',
      value: 'answer',
      valueType: 'string' as any,
      questionId: '123',
      order: 1,
      riskAdjustmentType: null,
      inSummary: false,
      summaryText: null,
    }],
    applicableIfType: null,
    applicableIfQuestionConditions: null,
  };
  const tree = create(
    <MockedProvider mocks={[]} store={mockStore({ locale })}>
      <ReduxConnectedIntlProvider>
        <ConnectedRouter history={history}>
          <RiskAreaQuestion
            visible={true}
            answerData={{ answers: [], oldAnswers: [], changed: false }}
            onChange={() => true}
            question={question}
            editable={true} />
        </ConnectedRouter>
      </ReduxConnectedIntlProvider>
    </MockedProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
