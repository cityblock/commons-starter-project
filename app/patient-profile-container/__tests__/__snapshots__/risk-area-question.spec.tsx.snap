// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a risk area question 1`] = `
<div
  className=""
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      Question Title
    </div>
    <div
      className={undefined}
    />
  </div>
  <div
    className="undefined"
  >
    <div
      className={undefined}
    >
      <div
        className=""
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <input
              checked={false}
              className={undefined}
              disabled={false}
              onClick={[Function]}
              type="radio"
              value="true"
            />
            <label
              className={undefined}
            />
          </div>
          <span
            className=""
            onClick={[Function]}
          >
            answer value
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
